src/compiler/checker.ts

```diff
@@ -336,6 +341,13 @@ namespace ts {
                         target[id] = source[id];
                     }
                     else {
+                        if (target === globals && source !== builtinGlobals) {

```

Comment:
This is a corner case, if the user ran into it, we just should let them know, i do not think we need to complicate our response here.